src/Libraries/Watch3DNodeModelsWpf/HelixWatch3DNodeViewModel.cs

```diff
@@ -74,9 +76,49 @@ protected override void OnModelPropertyChanged(object sender, PropertyChangedEve
             }
         }
 
+        protected override void OnNodePropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            var node = sender as NodeModel;
+            if (node == null)
+            {
+                return;
+            }
+
+            if(e.PropertyName == nameof(node.CachedValue))
+            {
+                var connected = watchModel.InPorts.SelectMany(p => p.Connectors.Select(c => c.Start.Owner));
+                if (!connected.Contains(node))
+                {
+                    return;
+                }
+
+                node.RequestVisualUpdateAsync(scheduler, engineManager.EngineController, renderPackageFactory, true, true);
+            }
+
+            if (e.PropertyName == "IsFrozen")
+            {
+                var visibleUpstream = new List<NodeModel>();
+                watchModel.VisibleUpstreamNodes(visibleUpstream);
+
+                if (!visibleUpstream.Contains(node))
+                { 
+                    return;
+                }
+
+                var gathered = new HashSet<NodeModel>();
+                node.GetDownstreamNodes(node, gathered);
+                SetGeometryFrozen(gathered);
+            }
+
+            node.WasRenderPackageUpdatedAfterExecution = false;
+        }
+
         protected override void PortDisconnectedHandler(PortModel obj)
         {
-            OnClear();
+            if (obj.PortType == PortType.Input  && watchModel == obj.Owner)
```

Comment:
when could it occur that obj is not a portModel of this node?

Reply:
In this case the PortConnected and PortDisconnected Handler are registered in the DefaultWatchViewModel as well.  So we need to filter out port events unrelated to this node.